scrapy/utils/curl.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import warnings
+from shlex import split
+
+from six.moves.http_cookies import SimpleCookie
+from six.moves.urllib.parse import urlparse
+from six import string_types, iteritems
+from w3lib.http import basic_auth_header
+
+
+class CurlParser(argparse.ArgumentParser):
+    def error(self, message):
+        error_msg = \
+            'There was an error parsing the curl command: {}'.format(message)
+        raise ValueError(error_msg)
+
+
+curl_parser = CurlParser()
+curl_parser.add_argument('url')
+curl_parser.add_argument('-H', '--header', dest='headers', action='append')
+curl_parser.add_argument('-X', '--request', dest='method', default='get')
+curl_parser.add_argument('-d', '--data', dest='data')
+curl_parser.add_argument('-u', '--user', dest='auth')
+
+
+safe_to_ignore_arguments = [
+    ['--compressed'],
+    ['-s', '--silent'],
+    ['-v', '--verbose'],
+    ['-#', '--progress-bar']
+]
+
+for argument in safe_to_ignore_arguments:
+    curl_parser.add_argument(*argument, action='store_true')
+
+
+def curl_to_request_kwargs(curl_args, ignore_unknown_options=True):
+    """Convert a cURL command syntax to Request kwargs.
+
+    :param str curl_args: string containing the curl command
+    :param bool ignore_unknown_options: If true, only a warning is emitted when
+    cURL options are unknown. Otherwise raises an error. (default: True)
+    :return: dictionary of Request kwargs
+    """
+
+    if isinstance(curl_args, string_types):
```

Comment:
Maybe we can omit the type check? According to the documentation a string if the only allowed value.

Reply:
Hi @Gallaecio, thank you for your feedback!
The `isinstance` is there since the first/original implementation was made and was there to support passing list arguments instead of a string. I’m not sure why that support was added, I think that maybe was added to support to use this function with already parsed arguments (python script to execute directly from the terminal?) but as we don’t need now I removed that part. Shouldn’t be difficult to implement if we need it in the future.